compel/src/lib/infect.c

```diff
@@ -860,7 +860,7 @@ static int parasite_memfd_exchange(struct parasite_ctl *ctl, unsigned long size,
 		goto err_cure;
 	}
 
-	ctl->remote_map = remote_mmap(ctl, NULL, size, remote_prot, MAP_FILE | MAP_SHARED, fd, 0);
+	ctl->remote_map = remote_mmap(ctl, (void *)ctl->ictx.user_set_remote_map, size, remote_prot, MAP_FILE | MAP_SHARED, fd, 0);
```

Comment:
If ctl->ictx.user_set_remote_map isn't zero, you probably need to consider setting MAP_FIXED_NOREPLACE.

Reply:
yes, thks. fixed it

Reply:
> If ctl->ictx.user_set_remote_map isn't zero, you probably need to consider setting MAP_FIXED_NOREPLACE.

yes, thks. fixed it